quipucords/api/serializers.py

```diff
@@ -39,3 +39,46 @@ def validate(self, attrs):
             raise ValidationError(_('ssh_keyfile, %s, is not a valid file'
                                     ' on the system.' % (ssh_keyfile)))
         return attrs
+
+
+class NetworkProfileSerializer(ModelSerializer):
+    """Serializer for the NetworkProfile model"""
+    class Meta:
+        model = NetworkProfile
+        fields = '__all__'
+
+    @staticmethod
+    def validate_name(name):
+        """Validate the name of the NetworkProfile."""
+        if not isinstance(name, str) or not name.isprintable():
+            raise ValidationError('NetworkProfile must have printable name.')
```

Comment:
It would be better to explicitly say what is printable, like characters in a-z, A-Z, and 0-9 for example. If I remember correctly from my past experience (~4-5 years ago) with Django this validation error's message can appear as a message on UI, that is why I think it would be better to improve it.